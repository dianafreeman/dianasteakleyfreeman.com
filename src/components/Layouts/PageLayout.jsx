/* eslint-disable no-return-assign */
import React from 'react';
import Helmet from 'react-helmet';
import PropTypes from 'prop-types';
import styled, { ThemeProvider } from 'styled-components';
import GlobalStyle from './GlobalStyle';
import { config, schemaOrgWebPage, breadcrumb } from '../../config/web/site';
import theme from '../../config/theme';

const PrimaryView = styled.main``;
const PageLayout = props => {
  const { pageTitle, children, backgroundColor = 'red', contentDescription } = props;
  const title = pageTitle || config.siteTitle;
  return (
    <ThemeProvider theme={theme}>
      <Helmet>
        <title>{title}</title>
        <link rel="apple-touch-icon" href="/favicons/apple-touch-icon.png" />
        <link rel="icon" type="image/png" sizes="32x32" href="/favicons/favicon-32x32.png" />
        <link rel="icon" type="image/png" sizes="16x16" href="/favicons/favicon-16x16.png" />
        <meta name="diana-m-steakley-freeman" content="Diana M. Steakley-Freeman" />
        <link rel="shortcut icon" href="favicon.ico" />
        <meta name="msapplication-TileColor" content={backgroundColor} />
        <meta name="msapplication-config" content="browserconfig.xml" />
        <meta name="description" content={contentDescription || config.description} />
        <meta name="image" content={config.image} />
        <meta property="og:locale" content={config.ogLanguage} />
        <meta property="og:site_name" content={config.ogSiteName} />
        <meta property="og:title" content={title} />
        <meta property="og:type" content="website" />
        <meta property="og:description" content={contentDescription || config.description} />
        <meta property="og:image" content={config.image} />
        <meta property="og:image:alt" content={contentDescription || config.description} />
        <meta name="twitter:card" content="summary_large_image" />
        <meta name="twitter:creator" content={config.userTwitter ? config.userTwitter : ''} />
        <meta name="twitter:title" content={title} />
        <meta name="twitter:description" content={contentDescription || config.description} />
        <meta name="twitter:image" content={config.image} />
        <meta name="twitter:image:alt" content={contentDescription || config.description} />
        <script type="application/ld+json">{JSON.stringify(schemaOrgWebPage)}</script>
        <script type="application/ld+json">{JSON.stringify(breadcrumb)}</script>
      </Helmet>
      <GlobalStyle theme={theme} />
      <PrimaryView id="primary" theme={theme}>
        {children}
      </PrimaryView>
    </ThemeProvider>
  );
};
export default PageLayout;

PageLayout.propTypes = {
  pageTitle: PropTypes.string.isRequired,
  children: PropTypes.arrayOf(PropTypes.node),
  theme: PropTypes.object.isRequired,
};
