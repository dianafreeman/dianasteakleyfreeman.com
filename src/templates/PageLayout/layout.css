body {
  padding: unset;
  margin: unset;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto,
    'Helvetica Neue', Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji',
    'Segoe UI Symbol';
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  position: relative;
}

main {
  padding: unset;
  margin: unset;
  height: 100vh;
}

.container {
  height: inherit;
  width: inherit;
}

.canvas-wrapper {
  height: 70%;
}

.content-wrapper {
  /* width: 50%; */
  margin: 1em;
}

.content {
  width: 100%;
  text-align: center;
  padding: 2em;
}

.navbar {
  width: 100%;
  top: 0;
  position: absolute;
  right: 0;
  left: 0;
}

/* DESKTOP */
@media screen and (min-width: 768px) {
  .container {
    display: flex;
    flex-direction: row-reverse;
  }
  .canvas-wrapper,
  .content-wrapper {
    width: 50%;
    height: 100%;
  }
}
