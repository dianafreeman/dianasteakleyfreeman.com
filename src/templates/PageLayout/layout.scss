@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@100;400;700;800;900&display=swap');

body {
  padding: unset;
  margin: unset;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto,
    'Helvetica Neue', Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji',
    'Segoe UI Symbol';
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  position: relative;
}

main {
  padding: unset;
  margin: unset;
  position: relative;
}

canvas {
  height: 100vh;
}

h1,
h2,
h3,
h4,
h5,
h6 {
  font-family: 'Poppins', sans-serif;
}

h1 {
  font-size: 50px;
  font-weight: 800;
}
h2 {
  font-size: 42px;
}
h3 {
  font-size: 38px;
}
h4 {
  font-size: 32px;
}
h5 {
  font-size: 28px;
}
h6 {
  font-size: 20px;
}
p {
  font-size: 18px;
}
