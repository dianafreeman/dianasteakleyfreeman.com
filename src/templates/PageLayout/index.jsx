import React, { Suspense, useState, useEffect } from 'react';

import { a } from '@react-spring/web';
import { useSpring } from '@react-spring/core';

import ThemeContext from '@context/ThemeContext';
import SEO from './SEO';

import './layout.scss';

const Navigation = () => {
  const { colors } = useTheme();
  return (
    <nav className="navbar navbar-dark navbar-expand-sm bg-dark py-2">
      <a className="navbar-brand" href="#">
        Navbar w/ text
      </a>
      <button
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarText"
        aria-controls="navbarText"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon"></span>
      </button>
      <div className="collapse navbar-collapse" id="navbarText">
        <ul className="navbar-nav mr-auto">
          <li className="nav-item active">
            <a className="nav-link" href="#">
              Home <span className="sr-only">(current)</span>
            </a>
          </li>
          <li className="nav-item">
            <a className="nav-link" href="#">
              Features
            </a>
          </li>
          <li className="nav-item">
            <a className="nav-link" href="#">
              Pricing
            </a>
          </li>
        </ul>
        <span className="navbar-text">Navbar text with an inline element</span>
      </div>
    </nav>
  );
};

function Layout({ children, postNode, postPath, postSEO }) {
  // ThreeJS Model Hover State
  const [hovered, setHovered] = useState(false);

  // ThreeJS Model PointerDown State
  const [down, setDown] = useState(false);

  // Global DarkMode State
  const [darkMode, setDarkMode] = useState(false);

  const colorMap = {
    dark: '#0d1f2b',
    light: '#f1e9df',
    gray: '#929093',
    primary: '#E8B059',
    primaryLight: '#F8C069',
    primaryDark: '#e09600',
    secondary: '#0094f1',
    secondaryLight: '#2fa4f4',
    secondaryDark: '#2277b1',
  };

  const dark = {
    background: colorMap.dark,
    text: colorMap.light,
    pointerLight: colorMap.secondaryLight,
    model: colorMap.secondaryDark,
  };
  const light = {
    background: colorMap.light,
    text: colorMap.dark,
    pointerLight: colorMap.primaryLight,
    model: colorMap.primaryDark,
  };

  const modelColor = (isDarkMode) => {
    if (isDarkMode) {
      return hovered ? colorMap.secondary : dark.model;
    }
    return hovered ? colorMap.primary : light.model;
  };

  // Colors
  const [colors] = useSpring(
    {
      background: darkMode ? dark.background : light.background,
      text: darkMode ? dark.text : light.text,
      pointerLight: darkMode ? dark.pointerLight : light.pointerLight,
      model: modelColor(darkMode),
      config: (n) =>
        n === 'wobble' && hovered && { mass: 2, tension: 1000, friction: 50 },
    },
    [darkMode, hovered],
  );

  // Springs for color and overall looks, this is state-driven animation
  // React-spring is physics based and turns static props into animated values
  const [springs] = useSpring(
    {
      wobble: down ? 0.95 : hovered ? 1.05 : 1,
      coat: darkMode && !hovered ? 0.04 : 1,
      env: darkMode && !hovered ? 0.4 : 1,
      ambient: darkMode && !hovered ? 1.5 : 0.5,
      config: (n) =>
        n === 'wobble' && hovered && { mass: 2, tension: 1000, friction: 30 },
    },
    [darkMode, hovered, down],
  );

  useEffect(() => {
    document.body.style.cursor = hovered
      ? 'none'
      : `url('data:image/svg+xml;base64,${btoa(
          `<svg width="32" height="32" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg"><circle cx="16" cy="16" r="10" fill="${
            darkMode ? colorMap.secondary : colorMap.primary
          }"/></svg>`,
        )}'), auto`;
  }, [hovered, darkMode]);

  return (
    <ThemeContext.Provider
      value={{
        down,
        setDown,
        darkMode,
        setDarkMode,
        hovered,
        setHovered,
        colors,
        springs,
      }}
    >
      <SEO postNode={postNode} postPath={postPath} postSEO={postSEO} />
      {/* <Navigation /> */}
      <a.main
        style={{
          background: colors.background,
        }}
      >
        {children}
      </a.main>
    </ThemeContext.Provider>
  );
}
export default Layout;
4;
