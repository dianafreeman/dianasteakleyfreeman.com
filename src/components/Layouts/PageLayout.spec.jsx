import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react'
import PageLayout from './PageLayout';

describe('PageLayout', () => {
  const props = {
    thisPageTitle:  "a test title",
    children:  [],
    backgroundColor:  "gray",
    contentDescription:  "desc"
  }

  it('should render without exploding', () => {
    expect(render(<PageLayout {...props} />)).not.to.throw
  })

  it('should match snapshot', () => {
    const tree = renderer.create(<PageLayout {...props} />).toJSON()
    jestExpect(tree).toMatchSnapshot();
  });
});