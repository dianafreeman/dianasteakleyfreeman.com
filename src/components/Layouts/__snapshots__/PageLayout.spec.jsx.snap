// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout should match snapshot 1`] = `
<main
  className=""
  id="primary"
/>
`;
